packages/dso-toolkit/src/components/badge/badge.mixins.scss

```diff
@@ -16,39 +17,42 @@
   text-align: center;
 
   & {
-    @include set-colors.apply(badge-variables.$default-bg-color, "badge");
+    @include _variant(badge-variables.$default-bg-color);
   }
 
   &.badge-info {
-    @include set-colors.apply(badge-variables.$info-bg-color, "badge");
+    @include _variant(badge-variables.$info-bg-color);
   }
 
   &.badge-primary {
-    @include set-colors.apply(badge-variables.$primary-bg-color, "badge");
+    @include _variant(badge-variables.$primary-bg-color);
   }
 
   &.badge-success {
-    @include set-colors.apply(badge-variables.$success-bg-color, "badge");
+    @include _variant(badge-variables.$success-bg-color);
   }
 
   &.badge-warning {
-    @include set-colors.apply(badge-variables.$warning-bg-color, "badge");
+    @include _variant(badge-variables.$warning-bg-color);
   }
 
   &.badge-error {
-    @include set-colors.apply(badge-variables.$error-bg-color, "badge");
+    @include _variant(badge-variables.$error-bg-color);
   }
 
   &.badge-outline {
-    @include set-colors.apply(colors.$wit, "badge");
-
-    // stylelint-disable-next-line no-duplicate-selectors -- needed to fix sass deprecation warnings: #2724
-    & {
-      border-color: colors.$grijs-90;
-    }
+    @include _variant(colors.$wit, colors.$grijs-90);
   }
 
   &.badge-attention {
-    @include set-colors.apply(badge-variables.$attention-bg-color, "badge");
+    @include _variant(badge-variables.$attention-bg-color);
+  }
+}
+
+@mixin _variant($background-color, $border-color: $background-color) {
+  @include set-colors.apply($background-color, "badge");
+
+  & {
+    border-color: $border-color;
```

Comment:
In deze mixin heb ik eningszins het oude gedrag van de set-colors mixin opnieuw geintroduceerd:

```scss
border-color: if($border-color, $border-color, $background-color);
```

Dit was natuurlijk een hele obscure routine die meer side-effects dan intended effects had. Mijns inziens is het terecht dat de set-colors mixin dit niet meer doet, maar dan moet het component (Badge) dat wel zelf doen.